store LastRemotePool of payload jobs (SOFTWARE-5438)

edquist opened the pull request:
My first whack at this...

get_collector_host() falls back on `condor_config_val COLLECTOR_HOST`, which is mostly borrowed from htcondor_configured()

edquist committed:
store LastRemotePool of payload jobs (SOFTWARE-5438)

and fall back on 'condor_config_val COLLECTOR_HOST'

...

get_collector_host() is mostly borrowed from htcondor_configured()

edquist commented:
oddly the CI failure is erroring on the `import classad` in the test module

timtheisen commented:
There was a problem with the HTCondor Python bindings in Pypi. I uploaded a corrected version to Pypi.

brianhlin commented on condor-ap/condor_meter:
```diff
@@ -893,6 +898,49 @@ def htcondor_configured():
+        cmd_stdout = cmd_stdout.strip()
+        if cmd.returncode != 0:
+            DebugPrint(-1, "WARNING: condor_config_val COLLECTOR_HOST "
+                           "returned a non-zero return code. Maybe the schedd "
```
```suggestion
                           "returned a non-zero return code. Maybe the schedd "
```

I think `condor_config_val` queries the `condor_master` daemon by default

edquist replied:
ah, good catch

edquist replied:
Updated

brianhlin commented on condor-ap/condor_meter:
```diff
@@ -893,6 +898,49 @@ def htcondor_configured():
+            return False
+        elif 'Not defined' in cmd_stdout:
+            DebugPrint(-1, "WARNING: COLLECTOR_HOST not set in the default "
+                           "condor schedd config. You may need to change or "
```
Same thing about `condor_master`

edquist replied:
Updated

brianhlin commented on condor-ap/condor_meter:
```diff
@@ -893,6 +898,49 @@ def htcondor_configured():
+            DebugPrint(-1, "WARNING: COLLECTOR_HOST not set in the default "
+                           "condor schedd config. You may need to change or "
+                           "reload the condor configuration or specify a "
+                           "different CondorScheddName in the Gratia config.")
```
```suggestion
                           "different CondorScheddName in the Gratia config.")
```

Hrm, I don't see where `CondorScheddName` comes into play here

edquist replied:
yeah this text was mostly copied from another function, apparently i only looked closely enough to update the referenced COLLECTOR_HOST attribute name

edquist replied:
Updated - dropped the reference to the CondorScheddName knob

brianhlin commented on condor-ap/condor_meter:
```diff
@@ -782,6 +782,11 @@ def classadToJUR(classad):
             r.Network(val, storageUnit='b', phaseUnit=networkPhaseUnit, metric=attr, description=attr)
     r.Network(total_network, storageUnit='b', phaseUnit=networkPhaseUnit, metric="total")
 
+    if not setIfExists(r.ExecutePool, classad, 'LastRemotePool', "Pool Host"):
+        collector_host = get_collector_host()
+        if collector_host:
+            r.ExecutePool(collector_host, "Pool Host from COLLECTOR_HOST")
```
Can we do some massaging here to handle

1. Multiple hosts
2. Sinful strings
3. Hosts with ports

I think this field should just be an XML list (as you suggsted) of FQDNS

edquist replied:
Sure thing!

edquist replied:
Ok I believe my latest push should handle all these cases, and include one or more `<ExecutePool>` tags in the XML

edquist committed:
fix up warning messages (SOFTWARE-5438)

edquist committed:
explicitly return None if no COLLECTOR_HOST is found (SOFTWARE-5438)

edquist committed:
parse out one or more host names from COLLECTOR_HOST (SOFTWARE-5438)

where COLLECTOR_HOST may have one or more of 'host', 'host:port', or
'<sinful>' -- and in the last case we parse out the 'alias' attribute.

edquist commented:
Ping @brianhlin, @djw8605 - I believe I addressed Brian's comments above and I don't think I was planning to add anything else here.  If we still want this feature the PR still needs a review to sign off on it.

djw8605 commented on condor-ap/condor_meter:
```diff
@@ -893,6 +897,68 @@ def htcondor_configured():
+        if (os.path.exists(condor_config_path) and
+            os.path.isfile(condor_config_path)):
+            args = [condor_config_path] + condor_config_val_args
+            cmd = subprocess.Popen(args, stdout=subprocess.PIPE)
```
Aren't we requiring htcondor python bindings already.  Why not use the python bindings to the the config value?

edquist replied:
That's a fair question - I am just following suit with the other parts of the probe that use the CLI tools.

djw8605 replied:
I vote using the python API.

brianhlin replied:
@djw8605 I agree but I'd prefer that we address that in a separate PR

djw8605 replied:
It can be iterative.  Use python API for this call now.  And a separate PR to fix everywhere else...

brianhlin replied:
Ya, I'd rather get this in. We'll tackle the CLI vs python bindings usage afterwards

djw8605 reviewed (commented):
Just 1 question.

brianhlin reviewed (approved):
LGTM
